libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -1444,3 +1453,27 @@ void NavEKF3_core::SampleDragData(const imu_elements &imu)
     }
 #endif // EK3_FEATURE_DRAG_FUSION
 }
+
+/*
+  get the earth mag field
+ */
+void NavEKF3_core::getEarthFieldTable(const Location &loc)
+{
+    table_earth_field_ga = AP_Declination::get_earth_field_ga(loc).toftype();
+    table_declination = radians(AP_Declination::get_declination(loc.lat*1.0e-7,
+                                                                loc.lng*1.0e-7));
+    have_table_earth_field = true;
+    last_field_update_ms = imuSampleTime_ms;
+}
+
+/*
+  check if we should update the earth field, we update it at 1Hz
+ */
+void NavEKF3_core::checkUpdateEarthField(void)
```

Comment:
This gets called from NavEKF3_core::UpdateFilter every IMU update, but only updates at 1Hz. We could move it into the bottom of if (runUpdates) {  so it gets called less often.

Reply:
fixed